source/MaterialXGenShader/HwShader.cpp

```diff
@@ -129,16 +129,8 @@ void HwShader::getTopLevelShaderGraphs(ShaderGenerator& shadergen, std::deque<Sg
 
 bool HwShader::hasSamplerNodes() const
 {
-    bool sampling = false;
-    for (SgNode* node : getNodeGraph()->getNodes())
-    {
-        if (node->hasClassification(SgNode::Classification::CONVOLUTION2D))
-        {
-            sampling = true;
-            break;
-        }
-    }
-    return sampling;
+    SgNodeGraph* graph = getNodeGraph();
+    return (graph && graph->hasClassification(SgNode::Classification::CONVOLUTION2D));
```

Comment:
One additional simplification you could do is to call hasClassification() directly on the shader. It's a method on the Shader base class that in turn calls getNodeGraph()->hasClassification(). An then you can remove the method HwShader::hasSamplerNodes().